src/compressor/CompressorClient.h

```diff
@@ -0,0 +1,44 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+#include "include/buffer_fwd.h"
+#include "CompressorRegistry.h"
+#include "auth/Auth.h"
+
+class CompConnectionMeta;
+class Connection;
+
+class CompressorClient {
+    CompressorRegistry comp_registry;
```

Comment:
i'd suggest move the private members at the end of class definition. see https://google.github.io/styleguide/cppguide.html#Declaration_Order